src/systems/velocity_demo/VelocityDemo.cc

```diff
@@ -0,0 +1,187 @@
+           << "Failed to initialize." << std::endl;
+    return;
+  }
+
+  // auto ptr = const_cast<sdf::Element *>(_sdf.get());
+
+  // Subscribe to commands
+  std::string topic{"/model/" + this->dataPtr->model.Name(_ecm) + "/cmd_vel"};
+  if (_sdf->HasElement("topic"))
+    topic = _sdf->Get<std::string>("topic");
+  this->dataPtr->node.Subscribe(
+    topic, &VelocityDemoPrivate::OnCmdVel, this->dataPtr.get());
+
+  ignmsg << "VelocityDemo subscribing to twist messages on [" << topic << "]"
+         << std::endl;
+}
+
+//////////////////////////////////////////////////
+void VelocityDemo::PreUpdate(const ignition::gazebo::UpdateInfo &_info,
+    ignition::gazebo::EntityComponentManager &_ecm)
+{
+  IGN_PROFILE("VelocityDemo::PreUpdate");
+
+  // \TODO(anyone) Support rewind
+  if (_info.dt < std::chrono::steady_clock::duration::zero())
+  {
+    ignwarn << "Detected jump back in time ["
+        << std::chrono::duration_cast<std::chrono::seconds>(_info.dt).count()
+        << "s]. System may not work properly." << std::endl;
+  }
+
+  // Nothing left to do if paused.
+  if (_info.paused)
+    return;
+
+  // update angular velocity of model
+  auto angularVel =
+    _ecm.Component<components::AngularVelocityCmd>(
+      this->dataPtr->model.Entity());
+
+  if (angularVel == nullptr)
+  {
+    _ecm.CreateComponent(
+      this->dataPtr->model.Entity(),
+      components::AngularVelocityCmd({this->dataPtr->angularVelocity}));
+    ignwarn << "Model angular velocity: "
+      << this->dataPtr->angularVelocity[0] << ", "
+      << this->dataPtr->angularVelocity[1] << ", "
+      << this->dataPtr->angularVelocity[2] << std::endl;
+  }
+  else
+  {
+    *angularVel = components::AngularVelocityCmd(
+      {this->dataPtr->angularVelocity});
+    ignwarn << "Model angular velocity: "
+      << this->dataPtr->angularVelocity[0] << ", "
+      << this->dataPtr->angularVelocity[1] << ", "
+      << this->dataPtr->angularVelocity[2] << std::endl;
+  }
+
+  // update linear velocity of model
+  auto linearVel =
+    _ecm.Component<components::LinearVelocityCmd>(
+      this->dataPtr->model.Entity());
+
+  if (linearVel == nullptr)
+  {
+    _ecm.CreateComponent(
+      this->dataPtr->model.Entity(),
+      components::LinearVelocityCmd({this->dataPtr->linearVelocity}));
+    ignwarn << "Model linear velocity: "
+      << this->dataPtr->linearVelocity[0] << ", "
+      << this->dataPtr->linearVelocity[1] << ", "
+      << this->dataPtr->linearVelocity[2] << std::endl;
+  }
+  else
+  {
+    *linearVel = components::LinearVelocityCmd(
+      {this->dataPtr->linearVelocity});
+    ignwarn << "Model linear velocity: "
+      << this->dataPtr->linearVelocity[0] << ", "
+      << this->dataPtr->linearVelocity[1] << ", "
+      << this->dataPtr->linearVelocity[2] << std::endl;
+  }
+}
+
+//////////////////////////////////////////////////
+void VelocityDemo::PostUpdate(const UpdateInfo &_info,
```

Comment:
It looks like the `PostUpdate` callback is not necessary, you could remove all lines related to it from this plugin.